### PR TITLE
[sil-combine] Enable convert_escape_to_no_escape -> thin_to_thick_function for ossa.

### PR DESCRIPTION
**Pull request by gottesmm**
Also begin a test file for casts specifically.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: test/SILOptimizer/sil_combine_casts.sil [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import Builtin
+
+class Klass {}
+
+sil @returnInt : $@convention(thin) () -> Builtin.Int32
+
+// CHECK-LABEL: sil @optimize_convert_escape_to_noescape :
+// CHECK: [[FN:%.*]] = function_ref @returnInt
+// CHECK: apply [[FN]]()
+// CHECK: } // end sil function 'optimize_convert_escape_to_noescape'
```
Comment (meg-gupta) [inline test/SILOptimizer/sil_combine_casts.sil:14]: don't we need a CHECK-NEXT or CHECK on the operand here ?

**Review by meg-gupta — APPROVED** [review]
lgtm

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Pull request merged**